sql/core/src/main/scala/org/apache/spark/sql/Observation.scala

```diff
@@ -78,20 +82,35 @@ class Observation(name: String) {
    * Get the observed metrics. This waits for the observed dataset to finish its first action.
    * Only the result of the first action is available. Subsequent actions do not modify the result.
    *
-   * @return the observed metrics as a [[Row]]
+   * @return the observed metrics as a `Map[String, Any]`
    * @throws InterruptedException interrupted while waiting
    */
   @throws[InterruptedException]
-  def get: Row = {
+  def get: Map[String, Any] = {
     synchronized {
       // we need to loop as wait might return without us calling notify
       // https://en.wikipedia.org/w/index.php?title=Spurious_wakeup&oldid=992601610
-      while (this.row.isEmpty) {
+      while (this.metrics.isEmpty) {
         wait()
       }
     }
 
-    this.row.get
+    this.metrics.get
+  }
+
+  /**
+   * (Java-specific)
+   * Get the observed metrics. This waits for the observed dataset to finish its first action.
+   * Only the result of the first action is available. Subsequent actions do not modify the result.
+   *
+   * @return the observed metrics as a `java.util.Map[String, Object]`
+   * @throws InterruptedException interrupted while waiting
+   */
+  @throws[InterruptedException]
+  def getAsJavaMap: java.util.Map[String, Object] = {
```

Comment:
Hmmm ... okay. this one might be one problem .. it requires to add another method which I actually would like to avoid ... but I guess it's fine